sqlpage/templates/table.handlebars

```diff
@@ -75,15 +85,37 @@
                             {{/if~}}
                         {{~/each~}}
                     </tr>
+
+                    {{! After this <tr> has been rendered, we need to check again if there is more work to do done }}
+                    {{#if (eq @row_index 0)}}
+                        {{#if ../first_row_is_footer}}
+                            {{! If the first row was the footer, 2 things must happen: we need to close the <tfoot> tag }}
+                            </tfoot>
+                            {{! and since it was not done before, the <tbody> tag needs to be added now, with a delayed </tbody> }}
+                            <tbody class="table-tbody list">{{#delay}}</tbody>{{/delay}}
+                        {{/if}}
+                    {{~/if~}}
                 {{/each_row}}
                 {{flush_delayed}}
-                {{#if (eq @row_index 0)}}
-                    <tbody class="table-tbody list">
-                        <tr>
-                            <td class="text-center">{{default empty_description 'No data'}}</td>
-                        </tr>
-                    </tbody>
-                {{/if}}
+
+                {{! If not enough rows were rendered, we need to place a 'No data' cell. "Not enough rows" depends on the footer settings }}
```

Comment:
The following `if` block should probably be modified: if can be re-written as a single `if` statement containing the same code. However, the condition is `(first_row_is_footer && row_index<=1) || (!first_row_is_footer && row_index == 0)` which I find unintelligible when written in handlebars, so I am not sure of any preference you might have for this style